iree/hal/local/local_executable.c

```diff
@@ -52,44 +52,52 @@ iree_hal_local_executable_t* iree_hal_local_executable_cast(
 iree_status_t iree_hal_local_executable_issue_call(
     iree_hal_local_executable_t* executable, iree_host_size_t ordinal,
     const iree_hal_executable_dispatch_state_v0_t* dispatch_state,
-    const iree_hal_vec3_t* workgroup_id, iree_byte_span_t local_memory) {
+    const iree_hal_executable_workgroup_state_v0_t* workgroup_state) {
   IREE_ASSERT_ARGUMENT(executable);
   IREE_ASSERT_ARGUMENT(dispatch_state);
-  IREE_ASSERT_ARGUMENT(workgroup_id);
+  IREE_ASSERT_ARGUMENT(workgroup_state);
   return ((const iree_hal_local_executable_vtable_t*)
               executable->resource.vtable)
-      ->issue_call(executable, ordinal, dispatch_state, workgroup_id,
-                   local_memory);
+      ->issue_call(executable, ordinal, dispatch_state, workgroup_state);
 }
 
 iree_status_t iree_hal_local_executable_issue_dispatch_inline(
     iree_hal_local_executable_t* executable, iree_host_size_t ordinal,
     const iree_hal_executable_dispatch_state_v0_t* dispatch_state,
-    iree_byte_span_t local_memory) {
+    uint32_t processor_id, iree_byte_span_t local_memory) {
   IREE_TRACE_ZONE_BEGIN(z0);
   // TODO(benvanik): annotate with executable name to calculate total time.
 
-  const iree_hal_vec3_t workgroup_count = dispatch_state->workgroup_count;
+  const uint32_t workgroup_count_x = dispatch_state->workgroup_count_x;
+  const uint32_t workgroup_count_y = dispatch_state->workgroup_count_y;
+  const uint32_t workgroup_count_z = dispatch_state->workgroup_count_z;
```

Comment:
(see my other comment) This then pulls the `uint16_t` out as a `uint32_t`?

Reply:
that's fine - the uint16_t is a compression for the in memory structure, not a requirement that it must be in a 16 bit register when used.